common-npm-packages/artifacts-common/credentialProviderUtils.ts

```diff
@@ -155,6 +172,42 @@ export function configureCredProviderForServiceConnectionFeeds(serviceConnection
     }
 }
 
+/**
+ * Configure the credential provider to provide credentials for a single service connections,
+ * using VSS_NUGET_EXTERNAL_FEED_ENDPOINTS to do so.
+ */
+export function configureSingleCredProviderForServiceConnectionFeeds(serviceConnections: ServiceConnection[]) {
+    var configuredEndpoints = tl.getVariable(CRED_PROVIDER_EXTERNAL_ENDPOINTS_ENVVAR);
+
+    if (serviceConnections) {
+        console.log(tl.loc('CredProvider_SettingUpForServiceConnections'));
+        
+        serviceConnections.map(authInfo => `${authInfo.packageSource.uri}`).forEach(serviceConnectionUri => console.log('  ' + serviceConnectionUri))
+        
+        // Ideally we'd also show the service connection name, but the agent doesn't expose it :-(
+        console.log();
+
+        const externalFeedEndpointsJson = buildExternalFeedEndpointsJson(serviceConnections);
+        const additionalCredentials = JSON.parse(externalFeedEndpointsJson).endpointCredentials;
+        const existingCredentials = configuredEndpoints ? JSON.parse(configuredEndpoints).endpointCredentials : [];
+        
+        for (const cred of existingCredentials) {
+            if (cred.hasOwnProperty("endpoint") && cred["endpoint"] === additionalCredentials[0]["endpoint"]) 
+            { 
+                // 
+                throw Error(tl.loc('CredProvider_Error_ServiceConnectionExists'));
```

Comment:
I threw here, but should the task warn instead of fail here?

Reply:
Just logging a warning message might hide the issue, but throwing an error would not allow any authenitcation to be added. I think we want the latter behavior and throwing would let the calling tasks catch and handle the error.

Reply:
As we discussed offline setting the task result to warntask might not be appropriate from a library. Throwing and catching in the calling tasks might ?

Reply:
This is a good idea. I have throws in here and will catch them in the task PR